src/Netscript/NetscriptHelpers.tsx

```diff
@@ -207,6 +216,45 @@ function runOptions(ctx: NetscriptContext, threadOrOption: unknown): CompleteRun
   return result;
 }
 
+function spawnOptions(ctx: NetscriptContext, threadOrOption: unknown): SpawnRunOptions {
```

Comment:
So, duplicating the options in `SpawnOptions` has some documentation benefits, but it also is duplication. I'm a bit leery of duplication, but I can see the argument there.

The amount of duplication here is going too far IMO. Probably you should be calling `threadOptions`, then doing additional casting/validation to get a `SpawnOptions`. Casting from a `ThreadOptions` to a `SpawnOptions` is gross though, and only needed because `SpawnOptions` isn't inheriting from `ThreadOptions` like it logically actually does.

Lets see what Snarling thinks.